Shooting loop [0.6.1]

jeger opened the pull request:
- different hit/shot suppliers are supported
- game's mechanics are now working as expected in functional requirements

Commit:
implement prototype, game with players

totine committed:
Add game loop for shooting

Commit:
Correct checkstyle violations

Commit:
Correct findbugs and checkstyle biolations

totine committed:
Refactor shooting game part

totine committed:
Fix compilation error

Commit:
remove versionsBackup files

Commit:
Change version

Commit:
Remove unecessary test file

jeger committed:
Change sout's to logger

KrzysztofDzioba reviewed (changes requested):
Game crashes after clicking "wyślij" in game window (NullPointerException). Please fix it

KrzysztofDzioba commented on client/src/main/java/gui/controllers/PlacementSceneController.java:
```diff
@@ -126,6 +126,7 @@ private void processMessagesFromServer() {
             @Override
             protected String call() {
                 String serverMessage = serverConnection.getMessage();
+                System.out.println(serverMessage);
```
use logger pls

AdrianHarenczyk replied:
We will add it later on. This issue will be tracked under number #37

KrzysztofDzioba commented on server/src/main/java/Main.java:
```diff
@@ -12,5 +12,6 @@ private Main() {
     public static void main(String[] args) {
         Server server = Server.createServer(SERVER_PORT);
         server.start();
+
```
blank line

AdrianHarenczyk replied:
Done

KrzysztofDzioba commented on server/src/main/java/connection/Server.java:
```diff
@@ -42,12 +44,15 @@ public void start() {
                 LOGGER.info(clientSocket.toString() + " connected");
                 MessagesFromClientHandler messagesFromClientHandler = new MessagesFromClientHandler(clientSocket, this);
+                fromClientHandlers.add(messagesFromClientHandler);
+                LOGGER.info(fromClientHandlers);
```
createServer() method is growing continously. You can think about extracting some parts of code into separate methods. It should be easier to read it later too.

totine replied:
It will be improved in next version, because there will be some new functionalities (e.g. multiroom)

KrzysztofDzioba commented on server/src/main/java/game/shooting/GameShootingPart.java:
```diff
@@ -0,0 +1,53 @@
+        this.playerBoardMatcher = playerBoardMatcher;
+    }
+
+    public static GameShootingPart create(Judge judge, PlayerBoardMatcher playerBoardMatcher,
```
Is there a need to declare it as static? All fields are already in this class provided by constructor. Nothing  use this method so far, but it is already public and static

totine replied:
It's a factory method - constructor is private

AdrianHarenczyk committed:
0-4-2-rebase:

Patch to the NPE deletion of buttons "Wyślij" and "Odbierz". There is no need for them now.

KrzysztofDzioba commented on server/src/main/java/game/shooting/GameShootingPart.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @Override
+    public void notifyObservers(ShotResults changes, ShootingBoard boardToShoot,
```
You dont have to pass boardToShoot parameter. You can get it from playerBoardMatcher based on given player

totine replied:
This approach is in new version, it will overwritten after rebase

KrzysztofDzioba commented on server/src/main/java/game/shooting/Judge.java:
```diff
@@ -0,0 +1,35 @@
+
+    }
+
+    public Player getCurrentPlayer() {
```
there is no need for it to be public for now

KrzysztofDzioba commented on server/src/main/java/game/shooting/Judge.java:
```diff
@@ -0,0 +1,35 @@
+        return players.getCurrentPlayer();
+    }
+
+    public boolean isEnd() {
```
same here

KrzysztofDzioba commented on server/src/main/java/game/shooting/ShotResults.java:
```diff
@@ -0,0 +1,26 @@
+        results = new HashMap<>();
+    }
+
+    public void put(Coordinates coordinates, FieldStateName stateName) {
```
put -> add ? For me it would be better, but thats only my opinion

KrzysztofDzioba commented on server/src/main/java/game/shooting/ShotResults.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class ShotResults {
```
javadocs?

AdrianHarenczyk replied:
This issue is tracked in number #38 and will be resolved later on.

KrzysztofDzioba commented on server/src/main/java/game/shooting/PlayerSwapper.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.List;
+
+
+public class PlayerSwapper {
```
docs?

AdrianHarenczyk replied:
This issue is tracked in number #38 and will be resolved later on.

KrzysztofDzioba commented on server/src/main/java/game/shooting/ObservableGame.java:
```diff
@@ -0,0 +1,11 @@
+
+
+
+public interface ObservableGame {
```
docs?

AdrianHarenczyk replied:
This issue is tracked in number #38 and will be resolved later on.

KrzysztofDzioba commented on server/src/main/java/game/shooting/Judge.java:
```diff
@@ -0,0 +1,35 @@
+import model.shooting.board.ShootingBoard;
+import model.shooting.field.HitResult;
+
+public class Judge implements BoardObserver {
```
docs?

AdrianHarenczyk replied:
This issue is tracked in number #38 and will be resolved later on.

KrzysztofDzioba commented on server/src/main/java/game/shooting/GameShootingPart.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class GameShootingPart implements ObservableGame {
```
All public classes/methods should have javadocs

AdrianHarenczyk replied:
This issue is tracked in number #38 and will be resolved later on.

KrzysztofDzioba commented on server/src/main/java/game/shooting/matchers/PlayerBoardMatcher.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class PlayerBoardMatcher {
```
docs also here and everywhere later

AdrianHarenczyk replied:
This issue is tracked in number #38 and will be resolved later on.

AdrianHarenczyk committed:
Patch blank line.

totine committed:
Fix access modifiers

totine committed:
Merge branch 'game-model-0-4-2-rebase' of https://github.com/totine/OStatki into game-model-0-4-2-rebase

AdrianHarenczyk commented:
Problematic parts of interface which were causing NPE are deleted now.

KrzysztofDzioba reviewed (changes requested):
Generally great work with boards staff and observers! Please provide more javadocs for easier reading, also I suggested small fixes

boczkas commented on server/src/main/java/connection/Server.java:
```diff
@@ -20,10 +21,12 @@
     private final ServerSocket serverSocket;
     private static final Logger LOGGER = LogManager.getLogger(Server.class);
     private List<MessagesToClientHandler> clientSocketList;
+    private List<MessagesFromClientHandler> fromClientHandlers;
```
messagesFromClientHandlers

AdrianHarenczyk committed:
Correct checkstyle errors.

boczkas commented on server/src/main/java/connection/Server.java:
```diff
@@ -20,10 +21,12 @@
     private Server(ServerSocket serverSocket, List<MessagesToClientHandler> list) {
         this.serverSocket = serverSocket;
         this.clientSocketList = list;
+        this.fromClientHandlers = new ArrayList<>();
```
new in constructor

boczkas commented on server/src/main/java/game/shooting/GameShootingPart.java:
```diff
@@ -0,0 +1,53 @@
+    private PlayerToShotSupplierMatcher shotSupplier;
+    private final PlayerBoardMatcher playerBoardMatcher;
+
+    private GameShootingPart(Judge judge, PlayerToShotSupplierMatcher hitSupplier,
```
Part of what?

boczkas commented on server/src/main/java/game/shooting/GameShootingPart.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    public void start() {
+        while (!judge.isEnd()) {
```
isRoundEnd/isGameEnd?

boczkas commented on server/src/main/java/game/shooting/GameShootingPart.java:
```diff
@@ -0,0 +1,53 @@
+    public void notifyObservers(ShotResults changes, ShootingBoard boardToShoot,
+                                Player currentPlayer) {
+        observers.forEach(observer -> observer.update(changes, boardToShoot, currentPlayer));
+
```
\n

boczkas commented on server/src/main/java/game/shooting/GameShootingPart.java:
```diff
@@ -0,0 +1,53 @@
+        observers.forEach(observer -> observer.update(changes, boardToShoot, currentPlayer));
+
+    }
+
```
\n

boczkas commented on server/src/main/java/game/shooting/Judge.java:
```diff
@@ -0,0 +1,35 @@
+        if (result.equals(HitResult.MISS)) {
+            players.swap();
+        }
+
```
\n

boczkas commented on server/src/main/java/game/shooting/Judge.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Override
+    public void update(ShootingBoard board, Coordinates coordinates, HitResult result) {
+        isEnd = board.areAllMastsDestroyed();
```
isGameEnd

boczkas commented on server/src/main/java/game/shooting/Judge.java:
```diff
@@ -0,0 +1,35 @@
+    boolean isEnd() {
+        return isEnd;
+    }
+
```
\n

boczkas commented on server/src/main/java/game/shooting/ObservableGame.java:
```diff
@@ -0,0 +1,11 @@
+import model.preparing.Player;
+import model.shooting.board.ShootingBoard;
+
+
```
\n

boczkas commented on server/src/main/java/game/shooting/PlayerSwapper.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.ArrayList;
+import java.util.List;
+
+
```
\n

boczkas commented on server/src/main/java/game/shooting/matchers/PlayerBoardMatcher.java:
```diff
@@ -0,0 +1,46 @@
+        return playerOwnBoards.get(player);
+    }
+
+    private static Map<Player, ShootingBoard> matchPlayersWithBoards(Player playerA, Player playerB,
```
unify convention of passing arguments to methods/constructors

boczkas commented on server/src/main/java/game/shooting/matchers/PlayerToShotSupplierMatcher.java:
```diff
@@ -0,0 +1,36 @@
+import model.Coordinates;
+import model.preparing.Player;
+
+
```
\n

boczkas commented on server/src/main/java/game/shooting/shotsuppliers/RandomShotSupplier.java:
```diff
@@ -0,0 +1,18 @@
+
+import java.util.Random;
+
+
```
\n

boczkas commented on server/src/main/java/game/shooting/shotsuppliers/ScannerShotSupplier.java:
```diff
@@ -0,0 +1,17 @@
+
+import model.Coordinates;
+
+
```
\n

boczkas commented on server/src/main/java/game/shooting/shotsuppliers/ScannerShotSupplier.java:
```diff
@@ -0,0 +1,17 @@
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+
+
```
\n

boczkas commented on server/src/main/java/game/shooting/shotsuppliers/ScannerShotSupplier.java:
```diff
@@ -0,0 +1,17 @@
+    public Coordinates getCoordinatesToShot() {
+        return Coordinates.create(scanner.nextInt(), scanner.nextInt());
+    }
+
```
\n

boczkas commented on server/src/main/java/model/preparing/Player.java:
```diff
@@ -0,0 +1,14 @@
+    public Player(String name) {
+        this.name = name;
+    }
+
```
\n

boczkas commented on server/src/main/java/model/shooting/board/ObservableBoard.java:
```diff
@@ -1,9 +1,10 @@
     void addObserver(BoardObserver observer);
-    void notifyObservers(Coordinates coordinates);
+    void notifyObservers(Coordinates coordinates, HitResult hitResult);
 
```
\n

boczkas commented on server/src/main/java/model/shooting/board/ShootingBoard.java:
```diff
@@ -8,19 +9,21 @@
     private ShootingBoardFields boardFields;
     private List<BoardObserver> observers;
+    private ShotResults changes;
 
```
\n

KrzysztofDzioba reviewed (approved):
ok

boczkas commented on server/src/main/java/model/shooting/board/ShootingBoard.java:
```diff
@@ -44,28 +47,39 @@ private void addObserversFromFleet(Fleet<PlacedShip> placedShipFleet) {
+        Field field = boardFields.get(coordinatesToDestroy);
         field.markDestroyed();
-        boardFields.getAllNeighboursCoordinates(coord).forEach(this::hit);
+        boardFields.getAllNeighboursField(coordinatesToDestroy).forEach(coordinates -> {
```
extract to createBuffor method

AdrianHarenczyk committed:
Overall code cleanup.

boczkas commented on server/src/main/java/model/shooting/board/ShootingBoardFields.java:
```diff
@@ -8,23 +8,23 @@
+    ShootingBoardFields(int rows, int cols) {
         this.rows = rows;
         this.cols = cols;
         coordinatesToFieldsMap = new HashMap<>();
```
new in constructor

boczkas commented on server/src/main/java/model/shooting/ship/ShootingShip.java:
```diff
@@ -11,18 +12,21 @@
 public class ShootingShip implements BoardObserver {
```
ShotShip

boczkas reviewed (approved):
Cosmetic changes
